sklearn/ensemble/_hist_gradient_boosting/splitting.pyx

```diff
@@ -381,18 +388,37 @@ cdef class Splitter:
             # map indices in left/right_indices_buffer back into
             # sample_indices. This also updates self.partition since
             # sample_indices is a view.
+            with gil:
+                print("\nsplit_indices")
+                print(f"n_threads = {n_threads}")
+                print(f"len(sample_indices) = {len(sample_indices)}")
+                print(f"sample_indices = {np.array(sample_indices)}")
             for thread_idx in prange(n_threads, schedule='static',
                                      chunksize=1):
+                with gil:
+                    print(
+                        f"\nthread_idx = {np.array(thread_idx)}"
+                        f"\noffset_in_buffers = {np.array(offset_in_buffers)}"
+                        f"\nleft_counts = {np.array(left_counts)}"
+                        f"\nleft_offset = {np.array(left_offset)}"
+                        f"\nleft_indices_buffer = {np.array(left_indices_buffer)}"
+                        f"\nright_counts = {np.array(right_counts)}"
+                        f"\nright_offset = {np.array(right_offset)}"
+                        f"\nright_indices_buffer = {np.array(right_indices_buffer)}",
+                        flush=True,
+                    )
```

Comment:
It looks like temporary changes for debugging have been committed.

Reply:
Uups. I was short in time. The "DEBUG" commit is reverted. I hope is fine again.